### PR TITLE
Use PSR-4 rather than PSR-0

### PR DESCRIPTION
**Pull request by iggyvolz**
https://www.php-fig.org/psr/psr-0/: 
> Deprecated - As of 2014-10-21 PSR-0 has been marked as deprecated. PSR-4 is now recommended as an alternative.

Used the PHPStorm refactoring tools as well as some manual labor to move all the classes to their correct location and use namespaces rather than underscores.  Also uses Composer's autoloader rather than the ad-hoc one.

Definitely needs some more testing to make sure it covers every possible class name access - there are some places where classes were accessed by strings but I _think_ I got all of those.

PDF renderer removed in the process, as per https://github.com/php/phd/pull/60#issuecomment-977993143.

Feel free to close if this is too ambitious or too much change all at once.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,14 +21,23 @@
         "php": ">=7.1.0",
         "ext-dom": "*",
         "ext-sqlite3": "*",
-        "ext-xmlreader": "*"
+        "ext-xmlreader": "*",
+        "ext-simplexml": "*"
```
Comment (iggyvolz) [inline composer.json:25]: SimpleXML is used in phpdotnet\phd\Package\IDE\API so I added that here

File: src/Options/OptionsInterface.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
-namespace phpdotnet\phd;
+namespace phpdotnet\phd\Options;
 
-interface Options_Interface {
+interface OptionsInterface {
```
Comment (iggyvolz) [inline src/Options/OptionsInterface.php:4]: Name here should be "Interface" which is not a valid class name

File: src/Format/_Abstract/XHTML.php [hunk 3]
```diff
@@ -1,7 +1,12 @@
 <?php
-namespace phpdotnet\phd;
+namespace phpdotnet\phd\Format\_Abstract;
```
Comment (iggyvolz) [inline src/Format/_Abstract/XHTML.php:2]: Abstract cannot be a namespace part - using _Abstract here

File: src/Package/IDE/API/_Function.php [hunk 4]
```diff
@@ -21,7 +24,7 @@
  * @license  http://www.opensource.org/licenses/bsd-license.php BSD Style
  * @link     https://doc.php.net/phd/
  */
-class Package_IDE_API_Function
+class _Function
```
Comment (iggyvolz) [inline src/Package/IDE/API/_Function.php:27]: Function is not a valid PHP class name

**Comment by Girgias** [comment]
See #52

**Comment by iggyvolz** [comment]
> See #52

Whoops not sure how I missed that, sorry for the noise.

**Pull request closed without merging**